superx/information_extractor.py

```diff
@@ -0,0 +1,201 @@
+            is_weighted = False
+            if item.find(bIsWeighted).text == 1:
+                is_weighted = True
+
+            unit_of_measure = 'none'
+            if is_weighted:
+                unit_of_measure = self.convert_unit_name(item.find('UnitQty').text)
+
+            products_list.append(Product(id=code, name=name, quantity=quantity, is_weighted=is_weighted, unit_of_measure=unit_of_measure))
+            branch_price_list.append(BranchPrice(item_code=code, branch_id=current_branch_id, price=price, update_date=update_date))
+
+        # add all the product info to the db and commit
+        db.session.add_all(products_list)
+        db.session.commit()
+
+        # add all the branch price info to the db and commit
+        db.session.add_all(branch_price_list)
+        db.session.commit()
+
+    def get_branch_id(self, tree_children_list):
+        """
+        Gets branch id from a parsed xml tree
+        :param tree_children_list: list of the children of the root
+        :return: the branch id
+        """
+        for child in tree_children_list:
+            if child.tag == 'StoreId':
+                return child.text
+
+    def convert_unit_name(self, unit_in_hebrew):
+        """
+        This method standardizes the unit of measurement
+        if the unit of measurement is not know, returns unknown
+        :param unit_in_hebrew: unit of measurement in hebrew
+        :return: standardized version of the unit or unknown if the unit is not known
+        """
+        unit_dict = {
+            'ק"ג': ['קילו', 'ק"ג', 'קילו', 'קילוגרמים'],
+            'גרם': ['גרם', 'גרמים'],
+            'ליטר': ['ליטר', 'ליטרים', 'ליטר    '],
+            'מ"ל': ['מיליליטרים', 'מ"ל', 'מיליליטר']
+            }
+
+        for unit in unit_dict.keys():
+            if unit_in_hebrew in unit:
+                return unit
+
+        return 'unknown'
+
+    def standardize_date(self, date):
+        """
+        This method standardizes the update date of the item
+        :param date: string representation of the date from the xml file
+        :return: standardized date as a string
+        """
+        # remove time from date
+        date = date[:10]
+        # set the format according to the supermarket
+        shuf_mega_format = '%Y-%m-%d'
+        victory_format = '%Y/%m/%d'
+        date_format = shuf_mega_format
+        if self.super_name is 'victory':
+            date_format = victory_format
+
+        new_date = datetime.strptime(date, date_format).date()
+        return new_date.__str__()
+
+    def update_super_name(self, url):
+        """
+        This method updates the super name according to the link
+        :param url: url of the current supermarket
+        """
+        if 'mega' in url:
+            self.super_name = 'mega'
+        elif 'shufersal' in url:
+            self.super_name = 'shufersal'
+        elif 'matrix' in url:
+            self.super_name = 'victory'
```

Comment:
Just a suggestion:
to make this more concise (and make it easier to add new supermarkets),
you can do either:
( if there is no need to change the name from matrix to victory)
```
supers = ['mega', 'shufersal', 'matrix']
for super in supers:
    if super in url:
        self.super_name = super
        break
or 
supers = {"mega": "mega", "shufersal": "shufersal", "matrix": "victory"}
for super in supers:
    if super in url:
        self.super_name = supers[super]
        break
```

Reply:
If you have multiple if statements in your code for variances in each store, you can make a dictionary of dictionaries, and define it all at the beginning. This will also make adding new supermarkets easier.
As an example:
```
supers = {"mega": {"store_name": "mega", "zip_link_prefix": "http://..." },
"shufersal": {"store_name": "shufersal"},
"matrix": {"store_name: "victory", "zip_link_prefix": "http://matrixcatalog.co.il/", "item": "Product", "bIsWeighted": "BisWeighted", "date_format": "%Y/%m/%d"}}
```
You could then use code such: 
`supers[self.super_name].get("item", "Item") #default to Item if it isn't specially defined`
and
`supers[self.super_name].get("zip_link_prefix", "")`

Reply:
I have added a dictionary of dictionaries, in the new version and will commit it shortly